contrib/win32/win32compat/w32fd.c

```diff
@@ -1052,7 +1053,9 @@ spawn_child_internal(char* cmd, char *const argv[], HANDLE in, HANDLE out, HANDL
 		error("failed to duplicate %s", cmd);
 		return ret;
 	}
-	
+	if (strstr(path, "system32\\cmd") || strstr(path, "ssh-shellhost"))
```

Comment:
What if you do:
bash.exe -c c:\windows\system32\cmd.exe /c "echo hi"

Reply:
Good catch.